### PR TITLE
Handle IgnoreAttribute in TestBrowser

### LINKED ISSUE #3260: Test browser does not support IgnoreAttribute

### PR DESCRIPTION
**Pull request by UselessToucan**
resolves #3260

### CODE REVIEW

File: osu.Framework.Tests/Visual/Testing/TestSceneIgnore.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace osu.Framework.Tests.Visual.Testing
+{
+    public class TestSceneIgnore : FrameworkTestScene
+    {
+        [Test]
+        [Ignore("test")]
+        public void IgnoredTest()
+        {
+            AddStep($"Throw {typeof(InvalidOperationException)}", () => throw new InvalidOperationException());
```
Comment (iiSaLMaN) [inline osu.Framework.Tests/Visual/Testing/TestSceneIgnore.cs]: nitpick but I would've added a false-assert step here: `AddAssert("fail due to not ignored", () => false);`

File: osu.Framework/Testing/TestBrowser.cs [hunk 2]
```diff
@@ -467,28 +467,31 @@ private void finishLoad(TestScene newTest, Action onCompletion)
                 if (name.StartsWith("Test"))
                     name = name.Substring(4);
 
-                if (m.GetCustomAttribute(typeof(TestAttribute), false) != null)
+                if (m.GetCustomAttribute(typeof(IgnoreAttribute), false) == null)
```
[the remark this answers is not in the document]
Comment (UselessToucan) [reply osu.Framework/Testing/TestBrowser.cs]: Theoretically, there should be an `else` branch, but at the moment there is no way to disable a `StepButton`.

[the remark this answers is not in the document]
Comment (smoogipoo) [reply osu.Framework/Testing/TestBrowser.cs]: An `else` branch to serve what purpose?

Either combining the condition or adding a secondary `if` is fine with me, just to reduce this nesting.

[the remark this answers is not in the document]
Comment (UselessToucan) [reply osu.Framework/Testing/TestBrowser.cs]: > An `else` branch to serve what purpose?

I believe ignored test should be added as a disabled `LabelStep` to the `StepsContainer`.

[the remark this answers is not in the document]
Comment (bdach) [reply osu.Framework/Testing/TestBrowser.cs]: > I believe ignored test should be added as a disabled `LabelStep` to the `StepsContainer`.

It's not the *worst* idea I guess, if a little superficial? Even if that was the case, it's a discussion for a separate PR (multiple PRs?), and "preparing" code like this for a theoretical PR that might not even be accepted is a little backwards to me.

**Pull request merged**